plugins/org.jboss.reddeer.eclipse/src/org/jboss/reddeer/eclipse/wst/xml/ui/tabletree/XMLSourcePage.java

```diff
@@ -62,52 +54,12 @@ protected XMLSourcePage(ITextEditor editor) {
 	 * @return the string
 	 */
 	public String evaluateXPath(String xPathExpression){
-		XPath xPath =  XPathFactory.newInstance().newXPath();
-
 		try {
-			Document doc = getDocument();
-			xPath.setNamespaceContext(new UniversalNamespaceResolver(doc));
-			return xPath.compile(xPathExpression).evaluate(doc);
+			XPathEvaluator xPathEvaluator = new XPathEvaluator(new ByteArrayInputStream(getText().getBytes()));
```

Comment:
There is a difference between getText and getAssociatedFile - what can see in the editor don't need to be in the file, e.g. unsaved state. So, I let the original logic here.